ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkFields/NetworkFields.tsx

```diff
@@ -70,13 +74,17 @@ const NetworkFields = ({
   const subnets: Subnet[] = useSelector(subnetSelectors.all);
   const { handleChange, setFieldValue, values } =
     useFormikContext<NetworkValues>();
-  const resetFollowingFields = (name: keyof NetworkValues) => {
+  const resetFollowingFields = (
+    name: keyof NetworkValues,
+    hasSubnet?: boolean
+  ) => {
     // Reset all fields after this one.
     const position = fieldOrder.indexOf(name);
     for (let i = position + 1; i < fieldOrder.length; i++) {
       let value = "";
       if (fieldOrder[i] === "mode") {
-        value = editing ? NetworkLinkMode.AUTO : NetworkLinkMode.LINK_UP;
+        value =
+          editing && hasSubnet ? NetworkLinkMode.AUTO : NetworkLinkMode.LINK_UP;
```

Comment:
`LINK_UP` is ["Unconfigured"](https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/store/utils/node/networking.ts#L472)